lib/src/_client.dart

```diff
@@ -306,9 +307,10 @@ class Client {
       if (trackTotalHits != null) 'track_total_hits': trackTotalHits,
       if (size != null) 'size': size,
       if (minScore != null) 'min_score': minScore,
+      if (knn != null) 'knn': knn,
```

Comment:
Would it be reasonable to add knn-helpers similar to these?
https://pub.dev/documentation/elastic_client/latest/elastic_client/Query-class.html#static-methods

Reply:
`Query` doesn't take classes either, it produces `Map` through static functions.

Reply:
I'm really not sure what you want here. Feel free to modify this yourself to meet your standards please.

This current implementation does what I need for my personal use case, so your request is a bit out of my scope, especially since I'm not familiar enough with the ElasticSearch API.